docs/java-run-your-first-app.md

```diff
@@ -0,0 +1,160 @@
+
+![Activity UI error details](/img/docs/web-ui-activity-error-info.png)
+
+[Timeout configurations](/docs/activities/#timeouts) and [retry policies](/docs/activities/#retries) for Activities are specified in the Workflow code as Activity options. This is another key value proposition that Temporal offers, as other modern applications implement timeout and retry logic within the business process itself. In our Workflow code you can see that we have specified a setStartToCloseTimeout for our Activities, and set a retry policy that tells the server to retry them up to 500 times. But we did that as an example for this tutorial, as Temporal automatically uses a default retry policy if one isn't specified!
+
+So, your Workflow is running, but only the `withdraw()` Activity succeeded. In any other application, the whole process would likely have to be abandoned and rolled back. But with Temporal, we can test the durable state of the Workflow again by stopping either the Temporal server, the Worker, or both and then starting them back up again. The Workflow will be in the same state!
+
+Now, pretend that you have a fix for the issue and you have commented the exception and saved the file. How do we deploy it to a process that is halfway complete? With Temporal, it is actually very simple: just restart the Worker!
+
+On the next scheduled attempt, the Worker will successfully execute the `deposit()` Activity and complete the Workflow. You have just fixed a bug "on the fly" with out losing the state of the Workflow.
+
+<img class="docs-image-centered" src={require('../static/img/docs/boost.png').default} />
+
+## Lore check &nbsp;&nbsp; ![](/img/docs/wisdom.png)
+
+Great work! In this tutorial we covered some of the practicalities of running a Temporal Workflow, code implementation, and concepts. Let's do a quick review to make sure you remember some of the more important pieces.
+
+![One](/img/docs/one.png) &nbsp;&nbsp; **What are three of Temporal's value propositions that we touched on in this tutorial?**
+
+1. Temporal gives you full visibility in the state of your Workflow and code execution.
+2. Temporal maintains the state of your Workflow, even through server outages and errors.
+3. Temporal makes it easy to timeout and retry Activity code using options that exist outside of your business logic.
+
+![Two](/img/docs/two.png) &nbsp;&nbsp; **How do you pair up Workflow code with a Worker that executes it?**
+
+Use the same Task Queue.
+
+![Three](/img/docs/three.png) &nbsp;&nbsp; **What do you have to do if you make changes to Activity code for a Workflow that is running?**
```

Comment:
Rephrase "what do you have to do to do we do so do"

Reply:
hmmm... i think i understand that...